### PR TITLE
Radix Button

### PR DESCRIPTION
**Pull request by istarkov**
## Description

Add radix button.

image

## Steps for reproduction

1. click button
2. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/react-sdk/src/component-renderer.tsx [hunk 1]
```diff
@@ -18,23 +21,55 @@ export const renderComponentTemplate = ({
...
-    ],
-    "base"
-  );
+
+  const template = metas.get(name)?.template ?? [
+    {
+      type: "instance",
+      component: name,
+      children: [],
+    } as EmbedTemplateInstance,
```
Comment (TrySound) [inline packages/react-sdk/src/component-renderer.tsx:39]: `const template: EmbedTemplate` instead?

File: packages/react-sdk/src/component-renderer.tsx [hunk 2]
```diff
@@ -18,23 +21,55 @@ export const renderComponentTemplate = ({
...
+  ];
+
+  if (template[0].type === "instance" && props !== undefined) {
+    template[0].props = Object.entries(props).map(([prop, value]) => {
+      if (typeof value === "string") {
+        return {
+          type: "string",
+          name: prop,
+          value: value,
+        };
```
Comment (TrySound) [inline packages/react-sdk/src/component-renderer.tsx:49]: One line?

File: packages/sdk-components-react-radix/src/theme/tailwind-colors.ts [hunk 3]
```diff
@@ -19,6 +19,15 @@ export const colors = {
...
   muted: `hsl(210 40% 96.1%)`,
+  primary: "rgb(15, 23, 42)",
+  primaryForeground: "hsl(210 40% 98%)",
+  destructive: "rgb(239, 68, 68)",
+  destructiveForeground: "rgb(248, 250, 252)",
+  accent: "rgb(241, 245, 249)",
+  accentForeground: "rgb(15, 23, 42)",
+  input: "rgb(226, 232, 240)",
+  secondary: "rgb(241, 245, 249)",
+  secondaryForeground: "rgb(15, 23, 42)",
```
Comment (TrySound) [inline packages/sdk-components-react-radix/src/theme/tailwind-colors.ts:30]: Maybe preserve hsl for everything?

**Review by TrySound — APPROVED** [review]

**Comment by kof** [comment]
shadcn button actually looks like: 


Screenshot 2023-08-11 at 20 55 38

**Comment by istarkov** [comment]
IMG_2465

just copied from his site, and looks like mine

File: packages/sdk-components-react-radix/src/button.ws.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  template: [
+    {
+      type: "instance",
+      component: "Button",
+      styles: [
+        // 'inline-flex items-center justify-center rounded-md text-sm font-medium
+        // ring-offset-background transition-colors
+        // focus-visible:outline-none focus-visible:ring-2
+        // focus-visible:ring-ring focus-visible:ring-offset-2
+        // disabled:pointer-events-none disabled:opacity-50',
```
Comment (TrySound) [inline packages/sdk-components-react-radix/src/button.ws.ts:59]: `'',` not needed

File: packages/sdk-components-react-radix/src/theme/tailwind-colors.ts (hunk 3 above)
Comment (istarkov) [reply packages/sdk-components-react-radix/src/theme/tailwind-colors.ts:30]: alpha for bg not working with hsl, some hsl -> rgb converter will be needed

**Comment by kof** [comment]
turned out shadcn safes changes to theme in local storage

**Review by kof — APPROVED** [review]

**Pull request merged**